kafka-client/src/main/java/dev/responsive/kafka/api/async/internals/AsyncThreadPool.java

```diff
@@ -153,6 +155,7 @@ public <KOut, VOut> void scheduleForProcessing(
       final AsyncUserProcessorContext<KOut, VOut> asyncProcessorContext,
       final AsyncProcessorMetricsRecorder processorMetricsRecorder
   ) {
+    maybeInitThreadPoolMetrics();
```

Comment:
I don't love doing this in #scheduleForProcessing...putting it in a separate #init method (maybe even waiting to pass the ResponsiveMetrics in until #init is called) seems slightly preferable to me. But honestly it's fine if you want to just merge it as-is

(my main concern is that this feels not very future proof, if we ever added another method to AsyncThreadPool that could potentially be invoked before #scheduleForProcessing is, and has some kind of metric to record)

Reply:
Oh I see what you mean - like an initMetrics that the processor calls from its init. I like that much better.

Reply:
exactly